numba/cuda/compiler.py

```diff
@@ -938,22 +1013,27 @@ def specialized(self):
 
     @property
     def definition(self):
+        warn('Use overloads instead of definition',
+             category=NumbaDeprecationWarning)
         # There is a single definition only when the dispatcher has been
         # specialized.
         if not self.specialized:
             raise ValueError("Dispatcher needs to be specialized to get the "
                              "single definition")
-        return next(iter(self.definitions.values()))
+        return next(iter(self.overloads.values()))
 
     @property
-    def _func(self, signature=None, compute_capability=None):
-        cc = compute_capability or get_current_device().compute_capability
-        if signature is not None:
-            return self.definitions[(cc, signature)]._func
-        elif self.specialized:
-            return self.definition._func
+    def definitions(self):
+        warn('Use overloads instead of definitions',
+             category=NumbaDeprecationWarning)
```

Comment:
`test:deprecation`

Reply:
Test added in https://github.com/numba/numba/pull/6432/commits/d9534677c20a7ab9b13e1a08df26024ed9438f8c